lib/prop/options.rb

```diff
@@ -10,15 +10,35 @@ def self.build(options)
       defaults = options.fetch(:defaults)
       result   = defaults.merge(params)
 
-      result[:key]       = Prop::Key.normalize(key)
-      result[:threshold] = result[:threshold].to_i
-      result[:interval]  = result[:interval].to_i
+      result[:key] = Prop::Key.normalize(key)
 
-      raise RuntimeError.new("Invalid threshold setting") unless result[:threshold] > 0
-      raise RuntimeError.new("Invalid interval setting")  unless result[:interval] > 0
+      result[:strategy] = if leaky_bucket.include?(result[:strategy])
+        Prop::LeakyBucketStrategy
+      elsif result[:strategy] == nil
+        Prop::IntervalStrategy
+      else
+        result[:strategy] # allowing any new/unknown strategy to be used
+      end unless proper_strategy.include?(result[:strategy])
 
+      result[:strategy].validate_options!(result)
       result
     end
 
+    def self.validate_options!(options)
```

Comment:
@grosser yes they are the same, but we still need this validation for LeakyBucketStrategy?